app/controllers/application_controller.rb

```diff
@@ -13,9 +13,15 @@ class ApplicationController < ActionController::API
 
   private
 
+  # This would get moved into the Insights Common optional_auth paths.
+  CATALOG_OPTIONAL_AUTH_PATHS = [
+    %r{\A/health\z},
+  ].freeze
+
   def with_current_request
     Insights::API::Common::Request.with_request(request) do |current|
-      if current.required_auth?
+      # TODO: move catalog_required_auth? paths to common gem once they are decided.
+      if current.required_auth? && catalog_required_auth?(URI.parse(current.original_url).path)
```

Comment:
I would suggest making a private method `required_auth?` in this controller and moving the logic there.  It would improve readability and setup the controller if we need to expand on this condition in the future.

I think @syncrou had some other thoughts about this that might negate my suggestion.

Reply:
I extracted the appropriate logic to the common gem, it looks a lot better now that the /health endpoint is handled by the `required_auth?` method.